README.md

```diff
@@ -35,3 +45,59 @@ spec:
     spec:
       bar: baz
 ```
+
+Such a CR can be applied using standard Kubernetes tooling such as [kubectl](https://kubernetes.io/docs/tasks/tools/#kubectl),
+or [ArgoCD](https://argoproj.github.io/cd/):
+
+```bash
+kubectl apply -f foo.yaml
+```
+
+### Observe an application instance
+
+The controller will continuously attempt to reconcile the desired state of the application instance
+and update the outcome of the reconciliation in the `status` field of the `AppInstance` custom resource.
+
+You can observe the `status` field of the `AppInstance` resource using standard Kubernetes tooling such as:
+
+```bash
+kubectl get appinstances foo -o json | jq .status
+```
+
+TIP: render logs in more readable format with:
+
+```bash
+kubectl get appinstances foo -o json | jq -r '.status.lastLogs|to_entries[] | "\(.key): \(.value)"'
+```
+
+### Creating a new package
+
+The `kubecfg pack` command can be used to take a jsonnet file and all its dependencies and push them
+all together as a bundle into an OCI artifact.
+
+```bash
+kubecfg pack ghcr.io/kubecfg/demo:v0.1.0 demo.jsonnet
+```
+
+### Installing packages manually
+
+You can run the same logic that the kubit controller does when rendering and applying a template by running
+the `kubit` CLI tool from your laptop:
+
+```bash
+kubit apply foo.yaml
+```
+
+Kubit is just a relatively thin wrapper on top of `kubecfg`.
+For increased compatibility, it uses standard `kubectl apply` to apply the manifests using more standard
+tooling rather than kubecfg's integrated k8s API.
+
+You can preview the actuall commands that `kubit` will run with:
+
+```bash
+kubit apply foo.yaml --dry-run=script
+```
+
+Other interesting optinos are `--dry-run=render` and `--dry-run=diff` which will respectively just render the YAML without applying it
+and rendering + diffing the manifests against a runnig application. This can be useful to preview effects of changes in the spec or
```

Comment:
```suggestion
and rendering + diffing the manifests against a running application. This can be useful to preview effects of changes in the spec or
```